topological_rviz_tools/CMakeLists.txt

```diff
@@ -1,108 +1,212 @@
-
-
-## Here we specify the list of source files.
-## The generated MOC files are included automatically as headers.
-set(SRC_FILES
-  src/topological_edge_tool.cpp
-  src/topological_node_tool.cpp
-  src/topological_map_panel.cpp
-  src/node_controller.cpp
-  src/topmap_manager.cpp
-  src/node_property.cpp
-  src/edge_property.cpp
-  src/pose_property.cpp
-  src/edge_controller.cpp
-  src/tag_controller.cpp
-  src/tag_property.cpp
+########################
+#
+#   FILE DECLARATION
+#
+
+# Declare the custom service files
+set(srv_files
+  "srv/AddEdge.srv"
 )
 
-## An rviz plugin is just a shared library, so here we declare the
-## library to be called ``${PROJECT_NAME}`` (which is
-## "rviz_plugin_tutorials", or whatever your version of this project
-## is called) and specify the list of source files we collected above
-## in ``${SRC_FILES}``.
-add_library(${PROJECT_NAME} ${SRC_FILES})
 
-add_dependencies(${PROJECT_NAME} ${PROJECT_NAME}_generate_messages_cpp)
 
-## Link the myviz executable with whatever Qt libraries have been defined by
-## the ``find_package(Qt4 ...)`` line above, or by the
-## ``set(QT_LIBRARIES Qt5::Widgets)``, and with whatever libraries
-## catkin has included.
-##
-## Although this puts "rviz_plugin_tutorials" (or whatever you have
-## called the project) as the name of the library, cmake knows it is a
-## library and names the actual file something like
-## "librviz_plugin_tutorials.so", or whatever is appropriate for your
-## particular OS.
-target_link_libraries(${PROJECT_NAME} ${QT_LIBRARIES} ${catkin_LIBRARIES})
-## END_TUTORIAL
-
-## Install rules
-
-install(TARGETS
-  ${PROJECT_NAME}
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+#######################
+#
+#   FILE GENERATION
+#
+
+# Generate the files
+rosidl_generate_interfaces(${PROJECT_NAME}
+  ${srv_files}
+  DEPENDENCIES ${srv_deps}
 )
 
-foreach (dir launch images conf)
-    install(DIRECTORY ${dir}/
-    DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/${dir})
-endforeach(dir)
 
 
-install(PROGRAMS
-  scripts/python_topmap_interface.py
-  DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-)
 
 
-install(FILES
-  plugin_description.xml
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
+#########################
+#
+#   BUILD RVIZ2 PLUGIN
+#
+
+# Group dependencies for easier referencing
+set(rviz_deps
+  ament_cmake
+  rclcpp
+  rclcpp_components
+  rviz_common
+  rviz_default_plugins
+  rviz_ogre_vendor
+  rviz_rendering
+  topological_navigation_msgs
+  pluginlib
+)
+
+## Specify the 'include' directories for the project
+#include_directories(
+#  include
+#  ${rclcpp_INCLUDE_DIRS}
+#  ${rviz_common_INCLUDE_DIRS}
+#  ${rviz_rendering_INCLUDE_DIRS}
+#  ${topological_navigation_msgs_INCLUDE_DIRS}
+#)
+#
+#
+#
+## Construct the library and define its dependencies
+## Declare files to be included under the plugin
+#set(rviz_files
```

Comment:
So, @Iranaphor this is where the C++ code needs to be re-added once it is changed to ROS2, right?

Reply:
Yes, by uncommenting the entire code block, the compiler crashes trying to build roscpp elements. More might be needed beyond to fully install the components but we can find that out as we progress further.